lib/plugins/sfDoctrinePlugin/lib/generator/sfDoctrineFormGenerator.class.php

```diff
@@ -554,44 +553,41 @@ public function underscore($name)
     return strtolower(preg_replace(array('/([A-Z]+)([A-Z][a-z])/', '/([a-z\d])([A-Z])/'), '\\1_\\2', $name));
   }
 
-  /**
-   * Get array of sfDoctrineColumn objects that exist on the current model but not its parent.
-   *
-   * @return array $columns
-   */
-  public function getColumns()
-  {
-    $parentModel = $this->getParentModel();
-    $parentColumns = $parentModel ? array_keys(Doctrine_Core::getTable($parentModel)->getColumns()) : array();
-
-    $columns = array();
-    foreach (array_diff(array_keys($this->table->getColumns()), $parentColumns) as $name)
+    /**
+     * Get array of sfDoctrineColumn objects that exist on the current model but not its parent.
+     *
+     * @return array $columns
+     */
+    public function getColumns()
     {
-      $columns[] = new sfDoctrineColumn($name, $this->table);
-    }
```

Comment:
What this `foreach` is doing is taking all the first-party properties of the model and adding them into `$columns` indiscriminantly.